internal/rbd/controllerserver.go

```diff
@@ -271,19 +271,27 @@ func checkValidCreateVolumeRequest(rbdVol, parentVol *rbdVolume, rbdSnap *rbdSna
 func (cs *ControllerServer) CreateVolume(
 	ctx context.Context,
 	req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
-	if err := cs.validateVolumeReq(ctx, req); err != nil {
+	var err error
+	if err = cs.validateVolumeReq(ctx, req); err != nil {
 		return nil, err
 	}
 
 	// TODO: create/get a connection from the the ConnPool, and do not pass
 	// the credentials to any of the utility functions.
-	cr, err := util.NewUserCredentials(req.GetSecrets())
+	secrets := req.GetSecrets()
```

Comment:
Because this code is duplicated in multiple places now, you should use a helper function for it. Maybe introduce
```
cr, err := util.NewUserCredentialsWithMigration(req.GetSecrets())
```
as a variation on `util.NewUserCredentials()`.

Reply:
converted to above model and all tests are passing :+1:  , ptal .. thanks @nixpanic